### PR TITLE
Added Azure.APIM.MultiRegionGateway

### LINKED ISSUE #1910: API Management instances should have multi-region deployment gateways enabled
# Rule request

## Suggested rule change

API Management instances should have multi-region deployment gateways enabled.

Azure API Management supports multi-region deployment. If the primary region goes offline, the secondary regions continue to serve API requests using the most recent gateway configuration. If a region goes offline, API requests are automatically routed around the failed region to the next closest gateway.

It is possible to enable or disable these gateways with the `properties.additionalLocations.disableGateway` property. The rule should ensure that if the property is declared, then it should have a value of false.

Reliability pillar for this one.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.ApiManagement/service]**

## Additional context

- [About multi-region deployment](https://learn.microsoft.com/azure/api-management/api-management-howto-deploy-multi-region#about-multi-region-deployment)
- [Azure template reference](https://learn.microsoft.com/azure/templates/microsoft.apimanagement/service)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #1910 

Added Azure.APIM.MultiRegionGateway rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

Hey Bernie, do you mind taking a look on this when you have the chance?

File: src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1 [hunk 1]
```diff
@@ -271,6 +271,14 @@ Rule 'Azure.APIM.MultiRegion' -Ref 'AZR-000340' -Type 'Microsoft.ApiManagement/s
     $Assert.GreaterOrEqual($TargetObject, 'properties.additionalLocations', 1).Reason($LocalizedData.APIMMultiRegion)
 }
 
+# Synopsis: API Management instances should have multi-region deployment gateways enabled.
+Rule 'Azure.APIM.MultiRegionGateway' -Ref 'AZR-000341' -Type 'Microsoft.ApiManagement/service' -If { (IsPremiumAPIM) -and (IsMultiRegion) } -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    $gateways = @($TargetObject.properties.additionalLocations |
+        Where-Object { $_.PSObject.Properties.Name -contains 'disableGateway' -and $_.disableGateway -ne $false })
+    
+    $Assert.Less($gateways, '.', 1).Reason($LocalizedData.APIMMultiRegionGateway)
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1:279]: I think we can simplify this to:

```suggestion
    $gateways = $PSRule.GetPath($TargetObject, 'properties.additionalLocations');
    foreach ($gateway in $gateways) {
        $Assert.HasDefaultValue($gateway, 'disableGateway', $False);
    }
```

File: docs/en/rules/Azure.APIM.MultiRegionGateway.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+API Management instances should have multi-region deployment gateways enabled.
+
+## DESCRIPTION
+
+Azure API Management supports multi-region deployment. If the *primary* region goes offline, the secondary regions continue to serve API requests using the most recent gateway configuration. If a *region* goes offline, API requests are automatically routed around the failed region to the next closest gateway.
+
+## RECOMMENDATION
+
+Enable multi-region deployment gateways.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.APIM.MultiRegionGateway.md:21]:
```suggestion
Consider enabling each regional API gateway location for multi-region redundancy.
```

File: docs/en/rules/Azure.APIM.MultiRegionGateway.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+# Multi-region deployment gateways
+
+## SYNOPSIS
+
+API Management instances should have multi-region deployment gateways enabled.
+
+## DESCRIPTION
+
+Azure API Management supports multi-region deployment. If the *primary* region goes offline, the secondary regions continue to serve API requests using the most recent gateway configuration. If a *region* goes offline, API requests are automatically routed around the failed region to the next closest gateway.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.APIM.MultiRegionGateway.md:17]: @BenjaminEngeset Each API gateway location is active-active, maybe lets change the working slightly to something like the following unless you have some alternative suggestions.

```suggestion
Azure API Management supports multi-region deployment.
Deploy API Management in multiple locations to:

- Provide active-active redundancy for API gateway requests across Azure regions.
- Serve the request from the closest API gateway region to the original request.

API gateways can be disabled to enabled you to test failover of your API workloads to another region.
When disabled, an API gateway will not route API traffic.
You should reenable API gateways after you have concluded failover testing to ensure that the API gateway is available for failover if another region becomes unavailable.

If a region goes offline, API requests are automatically routed around the failed region to the next closest gateway.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset looks good. I have a few suggestions. Let me know what you think or if you have any questions.

**Comment by BenjaminEngeset** [comment]
Great feedback, great to also learn from pull requests! I liked all the suggestions and I've implemented them.

Just a quick one before merge, does `GetPath` helper method return the property value such as the whole `additionalLocations` array or does it traverse thru the array and enumerate all the objectts one by one and return those?

**Review by BernieWhite — APPROVED** [review]
@BenjaminEngeset no worries. Great thanks for that. All good to merge.

**Pull request merged**